packages/synapse-react/src/ledger.ts

```diff
@@ -0,0 +1,280 @@
+        if (!isSupported) {
+          throw new Error('Ledger is not supported')
+        }
+      },
+
+      async connect({ chainId, withCapabilities } = {}) {
+        const chain = await getChain(config, chainId)
+        currentChainId = chain.id
+        try {
+          const eth = await this.getEth()
+          const appConfig = await eth.getAppConfiguration()
+          if (forceBlindSigning && appConfig.arbitraryDataEnabled === 0) {
+            throw new Error('Blind signing is not enabled on your Ledger device')
+          }
+
+          if (!onHidDisconnect) {
+            onHidDisconnect = this.onHidDisconnect.bind(this)
+            navigator.hid.addEventListener('disconnect', onHidDisconnect)
+          }
+
+          const { address } = await eth.getAddress(path, verifyAddress, false, currentChainId?.toString())
+          const _address = getAddress(address)
+          config.emitter.emit('connect', { accounts: [_address], chainId: currentChainId })
+          return {
+            accounts: (withCapabilities ? [{ address: _address, capabilities: {} }] : [_address]) as never,
+            chainId: currentChainId,
+          }
+        } catch (error) {
+          // Ledger device: UNKNOWN_ERROR (0x6511)
+          if (error instanceof TransportStatusError && error.statusCode === 25873) {
+            throw new Error('Open the Ethereum app on your Ledger device to continue')
+          }
+          throw error
+        }
+      },
+
+      async isAuthorized() {
+        const device = await findDevice(config)
+        const recentConnectorId = await config.storage?.getItem('recentConnectorId')
+
+        if (device && recentConnectorId === 'ledger') {
+          return true
+        }
+        return false
+      },
+
+      async disconnect() {
+        const eth = await this.getEth()
+        await eth.transport.close()
+        _eth = undefined
+        if (onHidDisconnect) {
+          navigator.hid.removeEventListener('disconnect', onHidDisconnect)
+          onHidDisconnect = undefined
+        }
+        await config.storage?.removeItem('ledgerDevice')
+        const device = await findDevice(config)
+        if (device) {
+          await device.forget()
+        }
+      },
+
+      async getClient(parameters?: { chainId?: number | undefined }) {
+        const chain = await getChain(config, parameters?.chainId ?? currentChainId)
+        const eth = await this.getEth()
+        const viemTransport = config.transports?.[chain.id]
+        if (!viemTransport) {
+          throw new Error('Viem Transport not found')
+        }
+        return createWalletClient({
+          transport: viemTransport,
+          key: 'ledgerWallet',
+          name: 'Ledger Wallet Client',
+          chain,
+          account: await ledgerToAccount({
+            transport: eth.transport,
+            accountIndex,
+            addressIndex,
+            changeIndex,
+          }),
+        })
+      },
```

Comment:
If changeAccount() is called, would not this return a wallet client for the original address, rather then the updated `path` ?